modyn/config/schema/pipeline/trigger/ensemble.py

```diff
@@ -51,15 +58,28 @@ def aggregate_decision_func(self) -> Callable[[dict[str, bool]], bool]:
     Field(discriminator="id"),
 ]
 
+# -------------------------------------------------------------------------------------------------------------------- #
+#                                                    EnsembleTrigger                                                   #
+# -------------------------------------------------------------------------------------------------------------------- #
+
 __TriggerConfig = ForwardRef("TriggerConfig", is_class=True)
 
 
 class EnsembleTriggerConfig(ModynBaseModel):
     id: Literal["EnsembleTrigger"] = Field("EnsembleTrigger")
 
-    policies: dict[str, __TriggerConfig] = Field(  # type: ignore[valid-type]
+    detection_interval_data_points: int = Field(
+        1000,
+        description=(
+            "The number of samples in the interval after which drift detection is performed. "
+            "Some policies don't require this time-dicretization. However some do, e.g. therefore, this parameter "
+            "is required also here."
+        ),
+        ge=1,
```

Comment:
I think here i am unsure why we need it on the ensemble level. If the time trigger triggers more frequently than e.g. the drift trigger, and we do majority vote, it is what it is. Why can't the drift trigger buffer this and run the detections, and if it has not seen enough data, just return False? I don't currently see why it's the job of the ensemble to care about this,  instead of just forwarding it to the sub triggers

Reply:
Let's maybe discuss that in person!

Reply:
Okay, I decided to go one step further and implement a more dynamic approach without ensemble level batching. I hope that's closer to what you imagined?